### ISSUE TITLE
Allow scanning in BLEINIT 2 (dual role)

### ISSUE THREAD

**Issue by mrx23dot** [espressif/esp-at#743]
### Is your feature request related to a problem?

When I don't advertise I should be able to scan for BLE devices.

Currently I have reboot to change between  AT+BLEINIT=2 and  AT+BLEINIT=1 just to start scanning.

**Comment by ustccw** [comment]
@Luxin812 PTAL

**Comment by Luxin812** [comment]
@mrx23dot This function will be supported recently, if there is an update, you will be notified as soon as possible.

**Comment by mrx23dot** [comment]
> @mrx23dot This function will be supported recently, if there is an update, we will let you know the result as soon as possible.

Is this fixed in v4? 
At least add a command to exit BLEINIT mode, so we could switch to other mode without rebooting.

**Comment by Luxin812** [comment]
@mrx23dot 
After the device initializes BLE, it can execute AT+BLEINIT=0 to deinit. AT+RST is not required. Please see the log below for specific operations. Can you provide your firmware and AT execution log?

```
[16:13:16:598] AT+RESTORE␍␊
[16:13:17:184] ␍␊
[16:13:17:184] OK␍␊
[16:13:18:395] ␍␊
[16:13:18:395] ready␍␊
[16:13:23:166] AT+BLEINIT=2␍␊
[16:13:23:253] ␍␊
[16:13:23:253] OK␍␊
[16:13:33:097] AT+BLEINIT=0␍␊
[16:13:33:336] ␍␊
[16:13:33:336] OK␍␊
[16:13:40:169] AT+GMR␍␊
[16:13:40:192] AT version:4.1.0.0-dev(f3c1850 - ESP32C3 - Feb  5 2024 10:09:06)␍␊
[16:13:40:192] SDK version:v5.0.4-dirty␍␊
[16:13:40:192] compile time(9061bdb3):Feb  5 2024 13:51:36␍␊
[16:13:40:192] Bin version:v3.2.0.0(MINI-1)␍␊
[16:13:40:192] ␍␊
[16:13:40:192] OK␍␊
```

**Comment by Luxin812** [comment]
After stopping ble broadcast and scanning (if there is a BLE connection, you need to execute the AT+BLEDISCONN command to disconnect the BLE connection) and then execute the AT+BLEINIT=0 command, the crash will not occur. 

In the future, the device will be able to scan as a BLE server.

**Comment by Luxin812** [comment]
Yes, the latest AT firmware on the master branch still has some known issues with the BLE server-side scanning functionality, and fixing these issues will take some time.

**Comment by JHedgesBartec** [comment]
@Luxin812 are you able to quantify how much time?

**Comment by JHedgesBartec** [comment]
@Luxin812 is there any progress on this feature request?

**Comment by Luxin812** [comment]
@JHedgesBartec 
Currently, the AT master has supported the BLE dual-role functionality. By setting [AT+BLEINIT=3](https://docs.espressif.com/projects/esp-at/en/latest/esp32/AT_Command_Set/BLE_AT_Commands.html#at-bleinit-bluetooth-le-initialization), the device can act as both a BLE server and a BLE client, and it also supports scanning capabilities. Please test with the latest master to confirm if this feature meets your requirements. The AT+BLEINIT=2 command no longer supports BLE scanning functionality.

**Comment by JHedgesBartec** [comment]
@Luxin812 thank you, will test and check.
